src/pagestyle.tex

```diff
@@ -0,0 +1,27 @@
+% Smaller margins
+\geometry{paper=a4paper}%, bottom=3cm, top=3cm, footnotesep=3cm}
+\pagestyle{fancy}
+
+% Remove indentation on linebreak
+\setlength\parindent{0pt}
+
+% Make top header line wider
+\addtolength\headwidth{\extramargins}
+% \fancyheadoffset{\dimexpr \extramargins/2\relax}
+
+
+%% Set header
+\makeatletter
+    \renewcommand{\headrulewidth}{.3pt} %% Change width of line under header
+    \lhead{\@coursename~--~\thetitle}
+    \cfoot{}
+    \rhead{\today}
+\makeatother
```

Comment:
No need to `\makeatother` when you're adding a new `\makeatletter` as next command?
Preferably just use `\makeatletter` on top (and even omit `\makeatother`), since when this is compiled to `.sty`, it's automatically `\makeatletter`.